src/ram-lib-mapping.ts

```diff
@@ -0,0 +1,451 @@
+    string,
+    JSONValue
+  > = createTypedMapFromEntries([
+    {
+      key: "auctionStartTime",
+      value: toJSONValue(event.params.timestampStart)
+    },
+    { key: "auctionEndTime", value: toJSONValue(event.params.timestampEnd) },
+    {
+      key: "auctionBasePrice",
+      value: toJSONValue(event.params.basePrice.toString())
+    },
+    {
+      key: "auctionNumTokensInAuction",
+      value: toJSONValue(event.params.numTokensInAuction.toString())
+    }
+  ]);
+
+  mergeProjectMinterConfigExtraMinterDetails(
+    projectMinterConfig,
+    auctionConfigUpdateDetails
+  );
+
+  projectMinterConfig.save();
+
+  // induce sync if the project minter configuration is the active one
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleNumTokensInAuctionUpdated(
+  event: NumTokensInAuctionUpdated
+): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // update num tokens in auction detail in project minter configuration extraMinterDetails json field
+  let auctionConfigUpdateDetails: TypedMap<
+    string,
+    JSONValue
+  > = createTypedMapFromEntries([
+    {
+      key: "auctionNumTokensInAuction",
+      value: toJSONValue(event.params.numTokensInAuction.toString())
+    }
+  ]);
+
+  mergeProjectMinterConfigExtraMinterDetails(
+    projectMinterConfig,
+    auctionConfigUpdateDetails
+  );
+
+  projectMinterConfig.save();
+
+  // induce sync if the project minter configuration is the active one
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleBidCreated(event: BidCreated): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const fullProjectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  const ramBidId = generateRAMMinterBidId(
+    event.address.toHexString(),
+    fullProjectId,
+    event.params.bidId.toString()
+  );
+
+  // @dev: a bid with this ID should not already exist for the RAM Minter
+  const bid = new Bid(ramBidId);
+  // Create new account entity if one for the bidder doesn't exist
+  const bidderAccount = new Account(event.params.bidder.toHexString());
```

Comment:
given the comment, should this be using a `load or create` style pattern?

Reply:
keeping this consistent with the account logic in the account logic in this repo - I think we do this because accounts only have an ID and derived fields so if it does exist we are overwriting the existing ID with the same value, saving us from having to call load